lib/pre2017.js

```diff
@@ -0,0 +1,115 @@
+const fetch = require('node-fetch')
+const cheerio = require('cheerio')
+const https = require('https')
+
+const PRE_URL = [
+  'https://codein.withgoogle.com/archive/2016',
+  'https://codein.withgoogle.com/archive/2015',
+  'https://www.google-melange.com/archive/gci/2014',
+  'https://www.google-melange.com/archive/gci/2013',
+  'https://www.google-melange.com/archive/gci/2012',
+  'https://www.google-melange.com/archive/gci/2011',
+  'https://www.google-melange.com/archive/gci/2010',
+]
+
+async function fetchText(url) {
+  const agent = new https.Agent({
+    rejectUnauthorized: false,
+  })
+  const res = await fetch(url, {
+    agent,
+  })
+
+  return await res.text()
+}
+
+async function fetchPre2017(url) {
+  const year = url.split('/').pop()
+  const html = await fetchText(url)
+  const json = await scrape(html, year)
+
+  return { html, json }
+}
+
+async function scrape(responseHTML, year) {
+  const html = cheerio.load(responseHTML)
+  let data = []
+
+  if (['2016', '2015'].includes(year)) {
+    const orgsList = html(`
+      section.home__winners
+      div.fixed-width
+      ul.no-style-list > li`)
+    data = orgsList.toArray().map(org => {
+      const orgHTML = cheerio.load(org)
+
+      return {
+        name: orgHTML('div.home__winner-box-header h3 a').text(),
+        winners: getUlList(
+          orgHTML,
+          'div.home__winner-box-content div:first-child ul li strong'
+        ),
+        finalists: getUlList(
+          orgHTML,
+          'div.home__winner-box-content div:last-child ul li'
+        ),
+      }
+    })
+  } else if (['2014', '2013', '2012'].includes(year)) {
+    const orgsList = html('div.org-winners-card')
+    data = orgsList.toArray().map(org => {
+      const orgHTML = cheerio.load(org)
+      const info = {
+        name: orgHTML('div.mdl-card__title a').text(),
+        winners: getUlList(
+          orgHTML,
+          'div.mdl-card__supporting-text ul:first-of-type li'
+        ),
+      }
+
+      if (year === '2014')
+        info.finalists = getUlList(
+          orgHTML,
+          'div.mdl-card__supporting-text ul:last-child li'
+        )
+
+      return info
+    })
+  } else if (['2011', '2010'].includes(year)) {
+    const main = html('div.main')
+    data = main.toArray().map(org => {
+      const orgHTML = cheerio.load(org)
+
+      return {
+        winners: getUlList(orgHTML, 'ul:first-of-type li'),
+        organizations: getUlList(orgHTML, 'ul:last-child li a'),
+      }
+    })
+  }
+
+  return {
+    year,
+    data,
+  }
+}
+
+function getUlList(html, selector) {
+  return html(selector)
+    .toArray()
+    .map(list =>
+      html(list)
+        .text()
+        .replace(/[\n\t]/g, '')
+    )
+}
+
+async function get() {
```

Comment:
Let's just back off to the beginning; you seem to not understand

export this function only, like before 

and see my another review on the test.